homeassistant/components/nexia/switch.py

```diff
@@ -20,6 +23,15 @@
 from .types import NexiaConfigEntry
 
 
+async def _stop_harmonizers(
+    _: Event, harmonizers: Iterable[NexiaRoomIQHarmonizer]
+) -> None:
+    """Await the shutdown methods in parallel when preparing to stop."""
+    async with asyncio.TaskGroup() as tg:
+        for harmonizer in harmonizers:
+            tg.create_task(harmonizer.async_shutdown())
```

Comment:
These tasks don't suspend to the event loop, so the overhead of task creation dominates for such short operations. Running them concurrently increases latency by ~2x. It's better to run them in series.